supervisor/dbus/udisks2/block.py

```diff
@@ -0,0 +1,231 @@
+            "options": bytearray(self.options),
+        }
+        for key in ("name", "passphrase-path"):
+            if not data[key]:
+                data.pop(key)
+        return data
+
+
+class UDisks2Block(DBusInterfaceProxy):
+    """Handle D-Bus interface for UDisks2 block device object.
+
+    http://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.Block.html
+    """
+
+    bus_name: str = DBUS_NAME_UDISKS2
+    properties_interface: str = DBUS_IFACE_BLOCK
+
+    @property
+    @dbus_property
+    def device(self) -> str:
+        """Return device file."""
+        return bytes(self.properties[DBUS_ATTR_DEVICE]).decode()
+
+    @property
+    @dbus_property
+    def id(self) -> str:
+        """Return unique identifer."""
+        return self.properties[DBUS_ATTR_ID]
+
+    @property
+    @dbus_property
+    def read_only(self) -> bool:
+        """Return whether device is read only."""
+        return self.properties[DBUS_ATTR_READ_ONLY]
+
+    @property
+    @dbus_property
+    def symlinks(self) -> list[str]:
+        """Return list of symlinks."""
+        return [
+            bytes(symlink).decode() for symlink in self.properties[DBUS_ATTR_SYMLINKS]
+        ]
+
+    @property
+    @dbus_property
+    def device_number(self) -> int:
+        """Return device number."""
+        return self.properties[DBUS_ATTR_DEVICE_NUMBER]
+
+    @property
+    @dbus_property
+    def configuration(
+        self,
+    ) -> dict[str, list[FstabConfigDetails | CrypttabConfigDetails]]:
+        """Return device configuration."""
+        configuration = defaultdict(list)
+        for type_, details in self.properties[DBUS_ATTR_CONFIGURATION]:
+            configuration[type_].append(
+                FstabConfigDetails.from_dict(details)
+                if "dir" in details
+                else CrypttabConfigDetails.from_dict(details)
+            )
+        return dict(configuration)
+
+    @property
+    @dbus_property
+    def size(self) -> int:
+        """Return size."""
+        return self.properties[DBUS_ATTR_SIZE]
+
+    @dbus_connected
+    async def add_configuration_item(
+        self,
+        type_: str,
+        details: FstabConfigDetails | CrypttabConfigDetails,
+        options: dict[str, Any] = None,
+    ):
+        """Add new configuration item."""
+        if not options:
+            options = {}
+        await self.dbus.Block.call_add_configuration_item(
+            (type_, details.to_dict()), options
+        )
+        await self.update()
+
+    @dbus_connected
+    async def remove_configuration_item(
+        self,
+        type_: str,
+        details: FstabConfigDetails | CrypttabConfigDetails,
+        options: dict[str, Any] = None,
+    ):
+        """Remove existing configuration item."""
+        if not options:
+            options = {}
+        await self.dbus.Block.call_remove_configuration_item(
+            (type_, details.to_dict()), options
+        )
+        await self.update()
```

Comment:
You may not need these `update` calls anymore. After #3872 all instances of `DBusInterfaceProxy` listen for property changes signals automatically and update their properties cache on change. So we should test this. If it fires that signal on change of properties after these method calls then calling `self.update` is redundant.